src/rez/package_test.py

```diff
@@ -114,42 +135,83 @@ def get_package(self):
         self.package = package
         return self.package
 
-    def get_test_names(self):
+    def get_test_names(self, run_on=None):
         """Get the names of tests in this package.
 
+        Args:
+            run_on (list of str): If provided, only include tests with run_on
+                tags within this list.
+
         Returns:
             List of str: Test names.
         """
         package = self.get_package()
-        return sorted((package.tests or {}).keys())
+
+        if package is None:
+            return []
+
+        if run_on:
+            tests_dict = dict(
+                (k, v) for k, v in package.tests.items()
+                if set(v.get("run_on", ["default"])) & set(run_on)
```

Comment:
I'm wondering if the default could be set in the schema? Is it something possible?

Reply:
It's probably possible, I thought about it, but if anything I want to avoid yet more schema code. The whole resource validation side of things needs a rethink - it works but there's a lot of redundancy, and I didn't want to add to that.

Reply:
Redundancy you say? You had copy/paste the schema addition in 3 different files. I cannot agree more on this. All good then.

Reply:
Yeah it's a pain, and getting into this opens a whole can of worms. There are 3 places where you need to make schema changes, but each of these schemas are subtly different. Until this schema/resource subsystem gets cleanup up (if that ever happens), I prefer to minimise additions to these schemas.